lib/_tls_wrap.js

```diff
@@ -846,10 +846,23 @@ Server.prototype.setOptions = function(options) {
     this.rejectUnauthorized = false;
   }
 
+
   if (options.pfx) this.pfx = options.pfx;
-  if (options.key) this.key = options.key;
+
+  if (!options.key) {
```

Comment:
Makes it sense to maybe add a type-check for `key` and `cert` (like it's done at https://github.com/nodejs/node/blob/master/lib/_tls_wrap.js#L837)?